File: requsetmanager/src/main/java/com/ut/requsetmanager/network/NetworkTask.java
package com.ut.requsetmanager.network;

import android.content.Context;

import com.ut.requsetmanager.callback.DataRequestCallback;
import com.ut.requsetmanager.callback.ProgressRequestCallback;
import com.ut.requsetmanager.entity.WebServiceParam;

import okhttp3.Call;

/**
 * Created by jianjianhong on 2017/11/18.
 */

public interface NetworkTask {
    <T> Call dataTask(Context context, WebServiceParam param, NetworkTaskImpl.DataTaskCallback<T> dataCallback);

    <T> Call downloadTask(Context context,WebServiceParam param, NetworkTaskImpl.ProgressTaskCallback<T> callBack);

    <T> Call uploadTask(Context context,WebServiceParam param, NetworkTaskImpl.ProgressTaskCallback<T> callBack);
}

File: networkrequest/src/main/java/com/anima/networkrequest/data/NetworkTask.kt
package com.anima.networkrequest.data

import com.anima.networkrequest.callback.DataDownloadCallback
import com.anima.networkrequest.data.okhttp.dataConvert.ResponseParser
import com.anima.networkrequest.entity.RequestParam

/**
 * Created by jianjianhong on 19-11-1
 */
interface NetworkTask {
    suspend fun <T> dataTask(param: RequestParam): ResponseParser
    suspend fun downloadTask(param: RequestParam, callBack: DataDownloadCallback): String
    suspend fun <T> uploadTask(param: RequestParam, callBack: DataDownloadCallback): ResponseParser
}